app/addons/documents/routes-index-editor.js

```diff
@@ -57,6 +58,10 @@ var IndexEditorAndResults = BaseRoute.extend({
     this.createDesignDocsCollection();
     this.addLeftHeader();
     this.addSidebar();
+
+    this.setComponent("#right-header", RightAllDocsHeader, {
+      database: this.database
+    });
```

Comment:
The all docs header includes the query options and the doc id search ahead. Its on the view pages and all docs page.